### PR TITLE
SPM-830 Fill lazy package summary and description

### PR DESCRIPTION
**Pull request by Josca**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: evaluator/evaluate_packages.go [hunk 1]
```diff
@@ -49,12 +49,12 @@ func lazySavePackages(tx *gorm.DB, names, evras []string) error {
 	}
 	defer utils.ObserveSecondsSince(time.Now(), evaluationPartDuration.WithLabelValues("lazy-package-save"))
 
-	name2ID, err := loadPkgName2IDMap(tx, names)
+	name2ID, nameID2Hashes, err := loadPackagesInfos(tx, names)
```
Comment (MichaelMraka) [inline evaluator/evaluate_packages.go:52]: Hmm, is name2id 1:1 or N:1 relation? If 1:1 can we save some memory by putting both into the same map?

File: evaluator/evaluate_packages.go [hunk 2]
```diff
@@ -63,34 +63,82 @@ func lazySavePackages(tx *gorm.DB, names, evras []string) error {
...
+
+	nameID2HashesCache = map[int]nameIDHashes{}
+	for _, hash := range hashes {
+		nameID2HashesCache[hash.NameID] = hash
+	}
+
+	return nameID2HashesCache, nil
+}
+
+func getPackageNameInfos(name string, name2ID map[string]int, hashes map[int]nameIDHashes) (bool, *nameIDHashes) {
```
Comment (semtexzv) [inline evaluator/evaluate_packages.go:96]: Please swap the order of arguments to be the same as is returned by map lookup

File: evaluator/evaluate_packages.go [hunk 3]
```diff
@@ -63,36 +63,63 @@ func lazySavePackages(tx *gorm.DB, names, evras []string) error {
 	return nil
 }
 
+type packageMetadata struct {
+	NameID          int
+	Name            string
```
Comment (MichaelMraka) [inline evaluator/evaluate_packages.go:68]: Is the name necessary here? It's a key in name2PackageMetadataCache map so you already know it when returning this struct from map. And if I see correctly it isn't used later anyway.

**Review by MichaelMraka — APPROVED** [review]
:+1:

**Review by semtexzv — APPROVED** [review]

**Pull request merged**